addons/website_forum/models/forum_forum.py

```diff
@@ -138,11 +138,18 @@ def _compute_teaser(self):
 
     @api.depends('post_ids')
     def _compute_last_post_id(self):
-        for forum in self:
-            forum.last_post_id = forum.post_ids.search(
-                [('forum_id', '=', forum.id), ('parent_id', '=', False), ('state', '=', 'active')],
-                order='create_date desc', limit=1,
-            )
+        self.last_post_id = False
+        forums_with_posts = self.filtered('post_ids')
```

Comment:
Wouldn't it be a lot more efficient to test against "total_posts", since that one is done using a nice read_group, rather than fetching all the related posts just to test that they're existing?

Alternatively you could just not check any of this, do your read group and default to False if the forum is not in the aggregated results (which is I feel what we usually do). Less code, less checks, KISS.